cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -57,6 +58,35 @@ class SimpleArray; // forward declaration
 namespace detail
 {
 
+template <typename T>
+concept IntegralType = std::is_integral_v<T>;
+
+template <typename T>
+concept ArithmeticType = std::is_arithmetic_v<T>;
+
+template <typename T>
+concept SimpleArrayType = requires(T t)
+{
+    typename T::value_type;
```

Comment:
Do you need this line?


```cpp
typename T::value_type;
```

Or did you intend to write:

```cpp
using value_type = typename T::value_type;
{t.data()}->std::convertible_to<value_type *>;
{t.size()}->std::convertible_to<size_t>;
```

Reply:
Thanks for catching this. Should be a mistake.